bundles/org.openhab.binding.tapocontrol/src/test/java/org/openhab/binding/tapocontrol/internal/discovery/TapoMDNS.java

```diff
@@ -0,0 +1,107 @@
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler class for TAPO Smart Home thing discovery over mDNS
+ *
+ * @author Christian Wild - Initial contribution
+ */
+@Component(configurationPid = "discovery.tapocontrol")
+@NonNullByDefault
+public class TapoMDNS implements MDNSDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(TapoMDNS.class);
+    private boolean isAutoDiscoveryEnabled = true;
+
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    @Modified
+    protected void modified(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    private void activateOrModifyService(ComponentContext componentContext) {
+        Dictionary<String, @Nullable Object> properties = componentContext.getProperties();
+        String autoDiscoveryPropertyValue = (String) properties
+                .get(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY);
+        if (autoDiscoveryPropertyValue != null && !autoDiscoveryPropertyValue.isBlank()) {
+            isAutoDiscoveryEnabled = Boolean.valueOf(autoDiscoveryPropertyValue);
+        }
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public String getServiceType() {
+        return "NULL";
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(ServiceInfo service) {
+        ThingTypeUID thingTypeUID = getThingType(service);
+        if (thingTypeUID != null) {
+            String id = service.getPropertyString(PROPERTY_FAMILY); // device id
+            return new ThingUID(thingTypeUID, id);
+        }
+        return null;
+    }
+
+    private @Nullable ThingTypeUID getThingType(final ServiceInfo service) {
+        String model = service.getPropertyString(PROPERTY_FAMILY); // model
+        logger.debug("Chromecast Type: {}", model);
+        if (model == null) {
+            return null;
+        }
+        return L510_THING_TYPE;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        if (isAutoDiscoveryEnabled) {
+            ThingUID uid = getThingUID(service);
+            if (uid != null) {
+                String host = service.getHostAddresses()[0];
+                int port = service.getPort();
+                logger.debug("Chromecast Found: {} {}", host, port);
```

Comment:
Chromecast ???

This implementation in tests is only to log data ? Because no `DiscoveryResult` is finally built ?

Reply:
yes this is only testing. There must be some discovery on original app because you don't need to enter the ip there.
But at the moment i don't know how it works so i'm triying some different discoveryservices to get it out sometime.